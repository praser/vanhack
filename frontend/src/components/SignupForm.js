import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import Input from '../components/Input';
import If from '../components/If';
import Loading from '../components/Loading';
import Container from '../components/Container';
import HTMLFormValidation from '../assets/js/html_form_validation';
import { signupRequest } from '../actions/index';

class SignupForm extends Component {
  constructor(props) {
    super(props);

    this.validate = this.validate.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  validate() {
    const form = new HTMLFormValidation(
      document.querySelector('form button'),
      [this.name, this.email, this.password, this.agreement],
    );
    return (form.handleButton());
  }

  handleSubmit(event) {
    event.preventDefault();
    this.props.signupRequest(
      this.name.value,
      this.email.value,
      this.password.value,
      this.agreement.value,
    );
  }

  render() {
    const size = 's12';
    return (
      <Container grid="s10 m8 l6 offset-s1 offset-m2 offset-l3" title="Create account">
        <form onSubmit={this.handleSubmit}>
          <div className="row">
            <Input
              label="Full name"
              name="name"
              size={size}
              msgError="Full name is a required field"
            >
              <input
                name="name"
                type="text"
                className="validate"
                required
                onChange={this.validate}
                ref={(input) => {
                  this.name = input;
                  return this.name;
                }}
              />
            </Input>

            <Input
              label="E-mail"
              name="email"
              size={size}
              msgError="The value you gave does not seems a valid e-mail address"
            >
              <input
                name="name"
                type="email"
                className="validate"
                required
                onChange={this.validate}
                ref={(input) => {
                  this.email = input;
                  return this.email;
                }}
              />
            </Input>

            <Input
              label="Password"
              name="password"
              size={size}
              msgError="Password is a required field"
            >
              <input
                name="name"
                type="password"
                className="validate"
                required
                onChange={this.validate}
                ref={(input) => {
                  this.password = input;
                  return this.password;
                }}
              />
            </Input>

            <div className="col s12">
              <input
                type="checkbox"
                name="agreement"
                id="agreement"
                className="validate"
                required
                onChange={this.validate}
                ref={(input) => {
                  this.agreement = input;
                  return this.agreement;
                }}
              />
              <label
                htmlFor="agreement"
                data-error="You must agree to signup"
              >
                I agree with terms of use.
              </label>
            </div>
          </div>
          <center className="row">
            <button
              className="btn waves-effect waves-light center"
              type="submit"
              name="action"
              disabled
            >
              Submit
              <i className="material-icons right">send</i>
            </button>
          </center>
        </form>
        <If test={this.props.isLoading}>
          <Loading />
        </If>
        <If test={Object.keys(this.props.user).length !== 0}>
          <Redirect to="/dashboard" />
        </If>
      </Container>
    );
  }
}

SignupForm.propTypes = {
  isLoading: PropTypes.bool,
  signupRequest: PropTypes.func.isRequired,
  user: PropTypes.object.isRequired,
};

SignupForm.defaultProps = {
  isLoading: undefined,
};

const mapStateToProps = state => ({
  user: state.login,
  isLoading: state.requestIsLoading,
});

const mapDispatchToProps = dispatch => ({
  signupRequest: (name, email, password, agreement) => (
    dispatch(signupRequest(name, email, password, agreement))
  ),
});

export default connect(mapStateToProps, mapDispatchToProps)(SignupForm);
