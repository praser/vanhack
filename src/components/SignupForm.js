import React, { Component } from 'react';
import { Button, Col, Icon, Input, Row} from 'react-materialize';

class SignupForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      name: '',
      email: '',
      password: '',
      agree: false
    }

    this.handleInput = this.handleInput.bind(this);
    this.handleCheckbox = this.handleCheckbox.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleInput(event) {
    const state = {};
    state[event.target.name] = event.target.value;
    this.setState(state);
  }

  handleCheckbox(event) {
    const state = {};
    state[event.target.name] = event.target.checked;
    this.setState(state);
  }

  handleSubmit(event) {
    event.preventDefault();
    console.log(this.state);
  }

  render() {
    return (
      <form className='col s12 m10 l8 offset-m1 offset-l2 z-depth-5' onSubmit={this.handleSubmit}>
        <Row>
          <Col>
            <h5 className="center">Create account</h5>
            <Input s={12} name='name' label='Full name' value={this.state.name} onChange={this.handleInput} validate>
              <Icon>person</Icon>
            </Input>

            <Input s={12} name='email' label="Email" type='email' value={this.state.email} onChange={this.handleInput} validate>
              <Icon>email</Icon>
            </Input>
            
            <Input s={12} name='password' label="Password" type='password' value={this.state.password} onChange={this.handleInput} validate>
              <Icon>lock</Icon>
            </Input>

            <Input name='agree' label='I have read and accept use terms.' type='checkbox' onChange={this.handleCheckbox} />
          </Col>
        </Row>
        <Row>
          <center>
            <Button waves='light' className='center blue-grey'>Create account</Button>
          </center>
        </Row>
      </form>
    )
  }
}

export default SignupForm