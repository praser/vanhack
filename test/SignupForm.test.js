import React from 'react';
import SignupForm from '../src/components/SignupForm';
import { wrap } from 'module';

describe('<SignupForm>', () => {
  it('must have a name input', () => {
    const wrapper = mount(<SignupForm />);
    expect(wrapper.find('input[name="name"]').length).toEqual(1);
  });

  it('name must be a text input', () => {
    const wrapper = mount(<SignupForm />);
    expect(wrapper.find('input[name="name"]').prop('type')).toEqual('text');
  })

  it('must have a email input', () => {
    const wrapper = mount(<SignupForm />);
    expect(wrapper.find('input[name="email"]').length).toEqual(1);
  })

  it('email must be a email input', () => {
    const wrapper = mount(<SignupForm />);
    expect(wrapper.find('input[name="email"]').prop('type')).toEqual('email');
  })

  it('must have a password input', () => {
    const wrapper = mount(<SignupForm />);
    expect(wrapper.find('input[name="password"]').length).toEqual(1);
  })

  it('password must be a password input', () => {
    const wrapper = mount(<SignupForm />);
    expect(wrapper.find('input[name="password"]').prop('type')).toEqual('password');
  })

  it('must have a agree input', () => {
    const wrapper = mount(<SignupForm />);
    expect(wrapper.find('input[name="agree"]').length).toEqual(1);
  })

  it('agree must be a checkbox input', () => {
    const wrapper = mount(<SignupForm />);
    expect(wrapper.find('input[name="agree"]').prop('type')).toEqual('checkbox');
  })

  it('must have only a button', () => {
    const wrapper = mount(<SignupForm />);
    expect(wrapper.find('button').length).toEqual(1);
  })

  it('renders SignupForm without crashing', () => {
    const wrapper = shallow(<SignupForm />);
    expect(wrapper).toMatchSnapshot();
  })
})