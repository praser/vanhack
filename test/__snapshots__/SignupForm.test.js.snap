// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupForm> renders SignupForm without crashing 1`] = `
<form
  className="z-depth-4"
  onSubmit={[Function]}
>
  <Row>
    <Col
      l={8}
      m={10}
      offset="m1 l2"
      s={12}
    >
      <h5
        className="center"
      >
        Create account
      </h5>
      <Input
        checked={false}
        label="Full name"
        name="name"
        onChange={[Function]}
        s={12}
        type="text"
        validate={true}
        value=""
      >
        <Icon>
          person
        </Icon>
      </Input>
      <Input
        checked={false}
        label="Email"
        name="email"
        onChange={[Function]}
        s={12}
        type="email"
        validate={true}
        value=""
      >
        <Icon>
          email
        </Icon>
      </Input>
      <Input
        checked={false}
        label="Password"
        name="password"
        onChange={[Function]}
        s={12}
        type="password"
        validate={true}
        value=""
      >
        <Icon>
          lock
        </Icon>
      </Input>
      <Input
        checked={false}
        label="I have read and accept use terms."
        name="agree"
        onChange={[Function]}
        type="checkbox"
      />
    </Col>
  </Row>
  <Row>
    <center>
      <Button
        className="center red lighten-3"
        node="button"
        waves="light"
      >
        Create account
      </Button>
    </center>
  </Row>
</form>
`;
